capi/src/lib.rs

```diff
@@ -190,6 +190,50 @@ pub unsafe extern "C" fn jieba_cut_for_search(
     }))
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn jieba_extract_textrank(
+    j: *mut CJieba,
+    sentence: *const c_char,
+    len: usize,
+    top_k: usize,
+    allowed_pos: *const *mut c_char,
+    allowed_pos_len: usize,
+) -> *mut CJiebaWords {
+    let jieba = j as *mut Jieba;
+    let c_str = CFixedStr::from_ptr(sentence, len);
+    // FIXME: remove allocation
+    let s = String::from_utf8_lossy(c_str.as_bytes_full());
+
+    let allowed_pos: Vec<String> = if allowed_pos_len == 0 {
+        Vec::new()
+    } else {
+        let mut v = Vec::with_capacity(allowed_pos_len);
+
+        for i in 0..allowed_pos_len {
```

Comment:
I think this loop can be simplified with [`slice::from_raw_parts`](https://doc.rust-lang.org/std/slice/fn.from_raw_parts.html)